### ISSUE TITLE
Addon: how to retreive a value from an internal variable ?

### ISSUE THREAD

**Issue by abouchama** [minishift/minishift#1960]
### General information
  * Minishift version: v1.11.0+4459917
  * OS: Linux 
  * Hypervisor: KVM 

### Steps to reproduce

Trying to create a new addon, the diffuculty that i have is that i run an 
`oc create -f oauthclient.yml `
in order to set oauthclient and after i use the token generated to create my application:

```
oc new-app dev \
-p OPENSHIFT_OAUTH_CLIENT_SECRET=$(oc sa get-token oauth-client)
```

the oc new-app take the following value $(oc sa get-token oauth-client) that's set on the oc create -f

Is there any way to make this as an internal variable
OPENSHIFT_OAUTH_CLIENT_SECRET=$(oc sa get-token oauth-client) ?

Thanks,

**Comment by budhrg** [comment]
> Is there any way to make this as an internal variable
OPENSHIFT_OAUTH_CLIENT_SECRET=$(oc sa get-token syndesis-oauth-client) ?

I think this type of requirement make sense.

**Comment by LalatenduMohanty** [comment]
Wondering how we have not come across this yet. cc @jorgemoralespou

**Comment by jorgemoralespou** [comment]
@LalatenduMohanty probably because current addons are simple. In my wrapper, addons are implemented as bash scripts andI use this functionality a lot, but also conditionals, which to me are more priority. But yes, definitely there's some script like features required for addons. 
Is there a place where requirements for addons can be seen in a roadmap/timeline type of way?

**Comment by LalatenduMohanty** [comment]
@jorgemoralespou You can see all add-ons related issues in https://github.com/minishift/minishift/labels/component%2Faddons
I agree that we have a substantial backlog of add-ons .

**Comment by LalatenduMohanty** [comment]
Resolved via #2448
